tests/Doctrine/Tests/Mocks/DriverResultMock.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Mocks;
+
+use ArrayIterator;
+use BadMethodCallException;
+use Doctrine\DBAL\Driver\Result;
+use PDO;
+use Traversable;
+
+use function array_values;
+use function count;
+use function current;
+use function next;
+use function reset;
+
+class DriverResultMock implements Result, ResultStatement
+{
+    /** @var list<array<string, mixed>> */
+    private $resultSet;
+
+    /**
+     * Creates a new mock statement that will serve the provided fake result set to clients.
+     *
+     * @param list<array<string, mixed>> $resultSet The faked SQL result set.
+     */
+    public function __construct(array $resultSet = [])
+    {
+        $this->resultSet = $resultSet;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchNumeric()
+    {
+        $row = $this->fetchAssociative();
+
+        return $row === false ? false : array_values($row);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchAssociative()
+    {
+        $current = current($this->resultSet);
+        next($this->resultSet);
+
+        return $current;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchOne()
+    {
+        $row = $this->fetchNumeric();
+
+        return $row ? $row[0] : false;
```

Comment:
`fetchOne` is about fetching one column, so it should look a bit more like `fetchAssociative` and not stop at 0 I think.

Reply:
> Returns the first value of the next row of the result

I think, my implementation is correct.

Reply:
Oh right I misread, it is very much correct 👍